src/routes/v1/auth.js

```diff
@@ -147,10 +149,11 @@ async function whoami(req, res) {
       email,
       contactNumber,
     } = await identityServices.usersService.findByGitHubId(userId)
-    return res.status(200).json({ userId, email, contactNumber })
+    res.status(200).json({ userId, email, contactNumber })
+    return statsService.countDbUsers()
```

Comment:
see above, this avoids blocking the client when we wanna collect logs

Reply:
we could, but this wouldn't be ideal. this is because `countDbUsers` involves a call to our database, which might take some time. 

the flow is as follows:
1. query db
2. submit metric to dd

for the first step, this requires a wait, which isn't ideal. because idw the user to pay for htis, i shifted it into the function itself rather than a middleware.

the alternative that i could do is to instead treat the route handler as middleware. this would allow us to abstract out the service call into the stats middleware but we do have to call `next` inside the route handler

Reply:
Hmm got it. I was thinking along the lines of spinning up a "gorountine" where you fire a async action in the background which branches off from the main-line of execution. Do you think thats possible haha

Else, i think the current implementation makes sense

The only irk I have is that an API handler is returning `statsService.countDbUsers()` which seems weird... A handler usally always returns with the res

what would happen if we do not await any of the actions - i.e. fire and forget?

Reply:
i tried it briefly on console and it worked (i.e. promise fulfilled without await) but yes do check/test the behaviour as well

Reply:
so the behaviour is 

```ts
const asyncFunc = async () => { 
  const needWait = await somePromise() // <-- waits here
  const notWaiting = somePromise() // <-- no wait
}
```

even if the function is declared async, because we don't `wait`, the main thread is free to continue